website/docs/guides/lifecycle.md

```diff
@@ -0,0 +1,172 @@
+---
+id: lifecycle
+title: Task Lifecycle
+---
+
+We have talked about how Effection operations are able to clean up after
+themselves, but how are they able to do this, and how can you implement your
+own operations which clean up after themselves?
+
+### Halt
+
+In order to understand the lifecycle of a Task, we must first understand the
+concept of halting a Task.
+
+In Effection, any task can be halted:
+
+``` javascript
+import { main } from 'effection';
+
+let task = main(function*() {
+  yield
+});
+
+task.halt();
+```
+
+Or:
+
+``` javascript
+main(function*() {
+  let task = yield spawn();
+  yield task.halt();
+});
+```
+
+Halting a Task means that the task itself is cancelled, it also causes any Task
+that has been spawned from the Task to be halted.
+
+### Return
+
+If a Task is driving a generator, we call `return()` on the generator. This behaves
+somewhat similarly to if you would put a `return` statement at the current yield point.
+
+``` javascript
+import { main } from 'effection';
+
+let task = main(function*() {
+  yield // we will `return` from here
+  console.log('we will never get here');
+});
+
+task.halt();
+```
```

Comment:
I'm guessing the `return()` you refer to is an internal thing?

Regarding the point you're making here, doesn't this apply to normal generator functions too (and not an effection-specific thing)?

Also do we need the `task.halt()` in this particular example?

Reply:
I was struggling to explain this in a good way. The end effect is similar to if you'd have replaced the `yield` with a `return` statement. So in the end, when we call `halt`, the function behaves somewhat similarly to this:

``` javascript
main(function*() {
  return;
  console.log('we will never get here');
});
```

So the reason we don't get to the console log is because we have "returned" from the generator function at the yield point.

If we don't halt the task then it's not very illustrative of the behaviour I'm trying to explain here.

I would love some ideas on how to explain this better though.

Reply:
I've rewritten this section slightly in an attempt to clarify. Let me know if it's more understandable this way.